### PR TITLE
Rgo2019 fix check resolve type

### LINKED ISSUE #769: check_resolve_type incorrect behaviour, incl errors
```py
bla = ds.Designator(initial_value=["bla1", "bla2"], resolve_type=[str])
```

```py
ds.check_resolve_type(bla, str, int)
(<type 'str'>, <type 'int'>)
<type 'tuple'>
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-392c7de3990e> in <module>()
----> 1 ds.check_resolve_type(bla, str, int)

/home/amigo/ros/kinetic/system/src/robot_smach_states/src/robot_smach_states/util/designators/checks.py in check_resolve_type(designator, *allowed_types)
     39         print allowed_types
     40         print type(allowed_types)
---> 41         real_allowed_type = allowed_types[0][0] #allowed_types is a list (because of the *).
     42 
     43         if not real_resolve_type == real_allowed_type:

TypeError: 'type' object has no attribute '__getitem__'
```

```py
ds.check_resolve_type(bla, str, [str])
(<type 'str'>, [<type 'str'>])
<type 'tuple'>
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-10-806ce8cf2884> in <module>()
----> 1 ds.check_resolve_type(bla, str, [str])

/home/amigo/ros/kinetic/system/src/robot_smach_states/src/robot_smach_states/util/designators/checks.py in check_resolve_type(designator, *allowed_types)
     39         print allowed_types
     40         print type(allowed_types)
---> 41         real_allowed_type = allowed_types[0][0] #allowed_types is a list (because of the *).
     42 
     43         if not real_resolve_type == real_allowed_type:

TypeError: 'type' object has no attribute '__getitem__'

```

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
fixes #769

### CODE REVIEW

File: robot_smach_states/src/robot_smach_states/util/designators/checks.py [hunk 1]
```diff
@@ -35,14 +36,24 @@ def check_resolve_type(designator, *allowed_types):
...
+            real_allowed_type = allowed_type[0] if isinstance(allowed_type, list) else allowed_type
+
+            if real_resolve_type == real_allowed_type:
+                correct_resolve_type = True
+                break
 
-        if not real_resolve_type == real_allowed_type:
-            raise TypeError("{0} resolves to {1} but should resolve to one of {2}".format(designator, designator.resolve_type, allowed_types))
+        if not correct_resolve_type:
+            raise TypeError("{0} resolves to {1} but should resolve to one of {2}".format(designator,
```
Comment (LoyVanBeek) [inline robot_smach_states/src/robot_smach_states/util/designators/checks.py]: Why not raise directly from line 39, instead of breaking?

Comment (MatthijsBurgh) [reply robot_smach_states/src/robot_smach_states/util/designators/checks.py]: Did you check after my last commit(511baf4a6b78e311b7aab5ea645343fe1130e3d6)?

Comment (LoyVanBeek) [reply robot_smach_states/src/robot_smach_states/util/designators/checks.py]: I did, but why split and set that bool and then read it later?

Comment (MatthijsBurgh) [reply robot_smach_states/src/robot_smach_states/util/designators/checks.py]: Because you need to check all allowed types for a good match.

**Pull request merged**